.github/workflows/repo-metrics-weekly.yml

```diff
@@ -0,0 +1,64 @@
+name: Weekly repo metrics
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 9 * * MON'
+
+permissions:
+  issues: write
+  pull-requests: read
+
+jobs:
+  build:
+    # this workflow will always fail in forks; bail if this isn't running in the upstream
+    if: github.repository == 'aws/aws-cdk'
+    name: metrics
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Get dates for last week
+      shell: bash
+      run: |
+        # Calculate the current day of the week (0=Sunday, 1=Monday, ..., 6=Saturday)
+        CURRENT_DAY_OF_WEEK=$(date +%u)
+
+        # Calculate the number of days to subtract to get to the previous Monday
+        DAYS_TO_MONDAY=$((CURRENT_DAY_OF_WEEK - 1))
+
+        # Calculate the date of the previous Monday
+        PREVIOUS_MONDAY=$(date -d "$DAYS_TO_MONDAY days ago" "+%Y-%m-%d")
+
+        # Calculate the date of the current Sunday (assuming today is not Sunday)
+        CURRENT_SUNDAY=$(date -d "$DAYS_TO_MONDAY days ago +6 day" "+%Y-%m-%d")
```

Comment:
I think this may not work since the `DAYS_TO_MONDAY` will return `0` if the script is running on monday. 
Moreover, since we already know the script will be executed only every monday based on the cron job scheduled, we can simplify as below to get last week (last monday to last sunday)
```
LAST_MONDAY=$(date -d "7 days ago" "+%Y-%m-%d")
LAST_SUNDAY=$(date -d "1 day ago" "+%Y-%m-%d")
```

Reply:
Updated